### PR TITLE
fix: Add missing inventory::submit

### PR DESCRIPTION
**Pull request by juchiast**
Signed-off-by: Duy Do <juchiast@gmail.com>

### CODE REVIEW

File: src/transforms/regex_parser.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ pub struct RegexParserConfig {
 }
 
 inventory::submit! {
-    TransformDescription::new::<RegexParserConfig>("regex_parser")
+    TransformDescription::new::<RegexParserConfig>("regex")
```
Comment (JeanMertz) [inline src/transforms/regex_parser.rs:38]: Is this correct? Everywhere else I still see it named `regex_parser`?

Comment (JeanMertz) [reply src/transforms/regex_parser.rs:38]: Ah I see `transform_type` returns `regex` as well. Should we make this consistent and rename it everywhere? For example, Serde still expects the `regex_parser` tag, it seems.

**Review by JeanMertz — CHANGES_REQUESTED** [review]
Good catch 👍

Out of curiosity, what happens when you forget to add a new component to the inventory?

File: src/transforms/regex_parser.rs (hunk 1 above)
Comment (juchiast) [reply src/transforms/regex_parser.rs:38]: I see the documentation uses `regex_parser`, so let's change `transform_type` to return `regex_parser`.

**Comment by juchiast** [comment]
> Out of curiosity, what happens when you forget to add a new component to the inventory? 

One thing it, `vector generate` won't work for those components.

**Review by JeanMertz — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Thanks! I don't suppose there is any way we could add a CI check for this? It seems like an easy thing to forget.

**Comment by binarylogic** [comment]
This and the whole encoding rules feature are easy to forget.

**Pull request merged**